pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -106,13 +107,19 @@ func (r *DeploymentReconciler) checkDeploymentExist(client client.Client) (const
 		return constants.CheckResultUnknown, nil, err
 	}
 	//existed, check equivalence
-	if semanticDeploymentEquals(r.Deployment, existingDeployment) {
+	if semanticDeploymentEquals(*r.Deployment, *existingDeployment) {
 		return constants.CheckResultExisted, existingDeployment, nil
 	}
+
+	diff, err := kmp.SafeDiff(r.Deployment.Spec, existingDeployment.Spec)
```

Comment:
we can use `SafeDiff` directly on line 110, the function might have option to exclude the replica field.

Reply:
what happens when user actually updates the deployment replica?

Reply:
> what happens when user actually updates the deployment replica?

If user modfiys the `isvc minReplicas`, the `hpa` will be updated, and then `hpa` will scale up/down the num of `Deployment`.

Reply:
@Iamlovingit I think this code can be simplified to the following on line 99 and remove line 103
```go
	ignoreFields := cmpopts.IgnoreFields(appsv1.DeploymentSpec{}, "Replicas")
	if diff, err := kmp.SafeDiff(r.Deployment.Spec, existingDeployment.Spec, ignoreFields); err != nil || diff == "" {
		return constants.CheckResultExisted, existingDeployment, err
	}
```